pkg/controller/buildrun/generate_taskrun.go

```diff
@@ -99,6 +91,18 @@ func generateTaskSpec(build *buildv1alpha1.Build, buildRun *buildv1alpha1.BuildR
 		Steps: []v1alpha1.Step{},
 	}
 
+	if build.Spec.BuilderImage != nil {
+		InputBuilderImage := v1alpha2.ParamSpec {
```

Comment:
This is worrying me a bit, because we are ending up leaking abstraction by mapping something in the `build` CR to a tekton object directly.

Reply:
Hi @sbose78 ,

Do you have any comment about this review?

Do you mean it is not good to map the `InputBuilderImage` as tekton object directly?

Thanks!

Reply:
I think this is fine!

Reply:
Ideally, the builder image should go into the Tekton object only if it was referred to in the BuildStrategy, right ?

Reply:
@sbose78 Yes, I think so!

Reply:
I don't think we had that 'check' in the first place, we can skip it for now. I'll let you two weigh in if that needs to be addressed later on :)

Reply:
I think we need to check it in the first place. Because for source-to-image buildStrategy, we must provide a builder image definition in build spec. Because for different source code languages, we need different builder images.